src/evict/evict_inline.h

```diff
@@ -588,30 +588,44 @@ __wti_evict_hs_dirty(WT_SESSION_IMPL *session)
       ((uint64_t)(conn->evict->eviction_dirty_trigger * bytes_max) / 100));
 }
 
+/*
+ * __evict_check_user_ok_with_eviction --
+ *     Check if the user wants to abort eviction in this session. #private
+ */
+static bool
+__evict_check_user_ok_with_eviction(WT_SESSION_IMPL *session, bool busy)
+{
+    /* Ask the user only if eviction is optional and it's a user session. */
+    if (busy || F_ISSET(session, WT_SESSION_INTERNAL))
```

Comment:
are internal session ever involved in app thread eviction? I thought it was only for app sessions